bokehjs/src/coffee/models/scales/linear_scale.coffee

```diff
@@ -0,0 +1,60 @@
+import {Scale} from "./scale"
+
+import * as p from "core/properties"
+
+export class LinearScale extends Scale
+
+  initialize: (attrs, options) ->
+    super(attrs, options)
+
+    @define_computed_property('state', @_state, true)
+    @add_dependencies('state', this, ['source_range', 'target_range'])
+    @add_dependencies('state', @source_range, ['start', 'end'])
+    @add_dependencies('state', @target_range, ['start', 'end'])
```

Comment:
@bryevdv, as long as you're cleaning things up, you could replace all those computed properties with straight implementation.